contrib/slurm/slurmexecutor.py

```diff
@@ -251,57 +283,120 @@ def run_slurm(benchmark, args, log_file):
                     benchmark.config.singularity,
                 ]
             )
-        srun_command.extend(args)
+        srun_command.extend(
+            [
+                "bash",
+                "-c",
+                f"{' '.join(args)} && echo 0 > exitcode || echo $? > exitcode",
+            ]
+        )
 
         logging.debug(
             "Command to run: %s", " ".join(map(util.escape_string_shell, srun_command))
         )
-        srun_result = subprocess.run(
-            srun_command,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-        )
-        logging.debug(
-            "srun: returncode: %d, output: %s",
-            srun_result.returncode,
-            srun_result.stdout,
-        )
-        jobid_match = jobid_pattern.search(str(srun_result.stdout))
-        if jobid_match:
-            jobid = int(jobid_match.group(1))
-        else:
-            logging.debug("Jobid not found in stderr, aborting")
-            stop()
-            return -1
+        jobid = None
+        while jobid is None:
+            with open(tmp_log, "w") as tmp_log_f:
+                srun_result = subprocess.run(
+                    srun_command,
+                    stdout=tmp_log_f,
+                    stderr=subprocess.STDOUT,
+                )
+
+            # we try to read back the log, in the first two lines there should be the jobid
+            with open(tmp_log, "r") as tmp_log_f:
+                first_lines = (
+                    tmp_log_f.readline(100).strip("\r\n")
+                    + "\\n"
+                    + tmp_log_f.readline(100).strip("\r\n")
```

Comment:
Why the `100` here?

Why the `strip()`? Python's "universal newlines mode" should be active by default and handle all kinds of newline characters.

Reply:
I am not sure why limiting this here is important? It seems like it also doesn't hurt, but it stands out and confuses the reader as to why it is necessary.

Thinking more about it, I would actually wonder whether it breaks something. What will calling `readline(100)` twice do if the first line is longer than the 100 characters? Will it actually read the second line or just the first (in two parts)? Also note that the limit for the first line cannot be effective if you want the second line, Python would have to read all of the first line anyway.

Bothering with `strip()`, concatenation with `\n`, and `str()` also do not seem necessary at all.

I would recommend something like:
```python
for line in itertools.islice(tmp_log_f, 2):
    jobid_match = jobid_pattern.search(line)
    ...
```

Reply:
The idea behind concetanation, etc, is that I want that to appear in the log (now only if the debug flag is given). For that, I need it to be single-line and reasonably long. However, that reason does not justify this level of complexity, so I'll rework this part, possibly with your recommendation. Thanks!